vault/logical_system.go

```diff
@@ -399,27 +401,57 @@ func (b *SystemBackend) handlePluginCatalogTypedList(ctx context.Context, req *l
 	if err != nil {
 		return nil, err
 	}
+	sort.Strings(plugins)
 	return logical.ListResponse(plugins), nil
 }
 
-func (b *SystemBackend) handlePluginCatalogUntypedList(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	pluginsByType := make(map[string]interface{})
+func (b *SystemBackend) handlePluginCatalogUntypedList(ctx context.Context, _ *logical.Request, _ *framework.FieldData) (*logical.Response, error) {
+	data := make(map[string]interface{})
+	var versionedPlugins []pluginutil.VersionedPlugin
 	for _, pluginType := range consts.PluginTypes {
 		plugins, err := b.Core.pluginCatalog.List(ctx, pluginType)
 		if err != nil {
 			return nil, err
 		}
 		if len(plugins) > 0 {
 			sort.Strings(plugins)
-			pluginsByType[pluginType.String()] = plugins
+			data[pluginType.String()] = plugins
 		}
+
+		versioned, err := b.Core.pluginCatalog.ListVersionedPlugins(ctx, pluginType)
+		if err != nil {
+			return nil, err
+		}
+
+		// Sort for consistent ordering
+		sort.SliceStable(versionedPlugins, func(i, j int) bool {
+			left, right := versionedPlugins[i], versionedPlugins[j]
+			if left.Type != right.Type {
+				return left.Type < right.Type
+			}
+			if left.Name != right.Name {
+				return left.Name < right.Name
+			}
+			if left.Version != right.Version {
+				return right.SemanticVersion.GreaterThan(left.SemanticVersion)
+			}
+
+			return true
```

Comment:
Since this is an implementation of `Less`, I think the default should be `false`, since at this point they two entries would be equal?
```suggestion
			return false
```

Reply:
Oops, thanks, good catch